src/models/users.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+class UserModel {
+  constructor(createId) {
+    this.users = [];
+    this.createId = createId;
+  }
+
+  getUsers() {
+    return this.users;
+  }
+
+  getUserById(userId) {
+    const foundUser = this.users.find(({ id }) => id === Number(userId));
+
+    return foundUser || null;
+  }
+
+  createUser(name) {
+    const newUser = {
+      id: this.createId(this.users),
+      name,
+    };
+
+    this.users.push(newUser);
+
+    return newUser;
+  }
+
+  removeUser(userId) {
+    this.users = this.users.filter(({ id }) => id !== Number(userId));
+  }
+
+  updateUser({ id, name }) {
```

Comment:
Add validation for the 'id' and 'name' parameters to ensure they're not empty or undefined, and return a 400 error if any of them is missing as specified in the task description.